Logging module

ryanrolds opened the pull request:
Adds logging module discussed in [#1390](https://github.com/adaptlearning/adapt_framework/issues/1399).

oliverfoster commented on src/core/js/logging.js:
```diff
@@ -0,0 +1,78 @@
+define(['core/js/adapt'], function(Adapt) {
```
could you put some whitespace into this file?

should be spaces 4 indentation, line breaks between function definitions etc

ryanrolds replied:
The 4 for indentation conflicts with: https://github.com/adaptlearning/documentation/blob/master/01_cross_workstream/style_guide.md which is linked in https://github.com/adaptlearning/adapt_framework/wiki/Contributing-code

oliverfoster replied:
Ignore the referenced/copied standard, it hasn't been formalized yet. It should say "spaces 4" indentation, no tabs, etc.

oliverfoster commented on src/core/js/logging.js:
```diff
@@ -0,0 +1,78 @@
+      this.debug('Logging config loaded');
+      this.trigger('log:ready');
+    },
+    debug: function(msg, data) {
```
these functions need their parameters to behaviour more like the ``console.log`` function.
essentially each argument forms part of the output string.

```javascript
    fatal: function() {
      this._log('fatal', [].array.slice.call(arguments, 0)));
    },
    _log: function(level, datas) {
      if (!this._config._isEnabled) {
        return;
      }

      if (this._levels[level] < this._levels[this._config._level]) {
        return;
      }

      if (this._config._console) {
        var log = [level.toUpperCase() + ':'];
        data && log.push.apply(log, datas);

        console.log.apply(console, log);
      }

      // Allow error reporting plugins to hook and report to logging systems
      this.trigger('log:' + level, datas);
    }
```

ryanrolds replied:
Will do.

ryanrolds commented:
* Fixed the indentation
* Adjusted log function arguments
* Query string override is now case-insensitive
* Added some blank lines
* Comments

Should be ready to review again.

ryanrolds commented:
IE8 has been EOL for a while now and as far as I'm aware is not supported on current MS OSs.

ryanrolds commented:
For ethical reasons I don't make changes to specifically support IE8, the browser is EOL and no longer receiving security updates, it's usage is negligent.

oliverfoster commented:
Fair enough. I'll see if someone else can take over.

oliverfoster commented:
functionality added on #1425 

thanks for doing this @ryanrolds 
nice bit of code
